crates/stackable-telemetry/src/instrumentation/axum/mod.rs

```diff
@@ -242,6 +286,28 @@ impl RequestExt for Request {
     }
 }
 
+fn server_host_to_tuple(
+    host: &str,
+    scheme: Option<&str>,
+) -> Result<(String, u16), ServerHostError> {
+    if let Some((host, port)) = host.split_once(':') {
+        // First, see if the host header value contains a colon indicating that
+        // it includes a non-default port.
+        let port: u16 = port.parse().context(ParsePortSnafu)?;
+        Ok((host.to_owned(), port))
+    } else {
+        // If there is no port included in the header value, the port is implied.
+        // Port 443 for HTTPS and port 80 for HTTP.
+        let port = match scheme {
+            Some("https") => DEFAULT_HTTPS_PORT,
+            Some("http") => DEFAULT_HTTP_PORT,
+            _ => return InvalidSchemeSnafu.fail(),
```

Comment:
...following on from a previous suggestion:

```suggestion
            Some(scheme) => return InvalidSchemeSnafu { scheme }.fail(),
            _ => return InvalidSchemeSnafu { scheme: String::new() }.fail(),
```

Reply:
Added in 47ffb40. I also turned it into an Option and combined it with `unwrap_or_default()`.

Reply:
Updated in fbb65ef2fb05ff501849c319e195bfc052231e54